### DISCUSSION TITLE
DAI 2024-2025 - Practical work 1 - Drin Racaj and Esteban Giorgis

### DISCUSSION THREAD

**Discussion by EstebanGiorgis** [heig-vd-dai-course/heig-vd-dai-course#365] [Show and tell]
1) call our application (SSH_Generate_with_Drin_And_Esteban) 2) the algorithm to encrypt 3) the path of the folder which will contain the 2 keys

**Comment by ludelafo** [comment]
We get the idea of your tool but could you provide us with a few more commands/examples?

The size of the project seems good to us.

**Reply by EstebanGiorgis** [reply]
Params : 
-e : protocole de chiffrement asymétrique (les valeurs possibles : rsa, ecc, dsa, elgamal, eddsa.
-p : path to folder (pour sauvegarder le fichier)
-f : nom du fichier

SSH_Generate_with_Drin_And_Esteban -e rsa -p /path/to/store -f filename
SSH_Generate_with_Drin_And_Esteban -e ecc
SSH_Generate_with_Drin_And_Esteban -p /path/to/folder -f filename
SSH_Generate_with_Drin_And_Esteban 


Si aucun paramètre n'est spécifié ou que des paramètres manquent, le CLI demandera à l'utilisateur le protocole de chiffrement, le dossier dans lequel sauvegarder le fichier, le nom du fichier. Si le path + nom de fichier existe déjà, nous demanderons à l'utilisateur s'il veut remplacer le fichier.

Si le temps nous le permet, nous ajouterons la possibilité de charger un fichier texte contenant un algorithme (asymétrique) écrit en java pour le paramètre  -e qui sera compilé et exécuté pendant le runtime.

**Reply by ludelafo** [reply]
Merci pour votre retour.

Donc si je comprends bien, votre CLI a pour but de générer des clés cryptographiques (une paire de clés privée + publique) portant un certain nom et pouvant être stocké dans un chemin particulier (un dossier spécifié par l'utilisateur par exemple) en utilisant un algorithme de chiffrement (RSA, AES, etc.) pour pouvoir utiliser ces clés dans d'autres projets, est-ce que j'ai bien compris ?

Vous auriez donc une commande principale ou une sous-commande `generate` qui vous permettrait de faire ça, est-ce correct ?

Ça me semble OK dans le temps imparti mais peut-être que si vous avez le temps, je partirais plus sur deux sous-commandes `encrypt` et `decrypt` qui pourraient prendre en paramètre une des deux clés (privée ou public), un message texte, et chiffrer/déchiffrer ce message texte avec la clé souhaitée, ce qui me semble plus facile à réaliser et plus utile pour une utilisation globale.

Qu'est-ce que tu en penses @hadrylouis ?

**Reply by EstebanGiorgis** [reply]
Votre idée semble meilleure et surtout plus "utile", donc si je résume : En plus d'avoir une commande qui permettrait de générer des clés privées et publiques (avec une commande "generate") à un emplacement donné, avoir aussi deux autres commandes pour chiffrer/déchiffrer un fichier, à savoir "encrypter" et "décrypter" qui permettrait comme leur nom l'indique d'encrypter ou de déchiffrer un message texte à l'aide de clé que l'on aurait généré grace a la première commande. Avons-nous bien saisi votre idée ?

**Comment by EstebanGiorgis** [comment]
Comme demandé pour le rendu de ce projet:

- Voila le lien de notre repo github : https://github.com/EstebanGiorgis/DAI_Project_1

- Et voila le hash de notre dernier commit : 1b484d8a03d3cae30993c2f7b414fe3dd21e097a


PS : Nous vous avons aussi invité en tant que collaborateur de notre projet, car notre repo est privé

**Reply by ludelafo** [reply]
Merci pour l'invitation @EstebanGiorgis et @drinracaj,

Est-il possible d'inviter @GeraudSilvestri pour la correction ?

Merci et bonne journée,
Ludovic
